gax/src/test/java/com/google/api/gax/batching/v2/BatcherImplTest.java

```diff
@@ -246,19 +285,157 @@ public void testWhenThresholdIsDisabled() throws Exception {
             .build();
     underTest =
         new BatcherImpl<>(
-            SQUARER_BATCHING_DESC_V2, callLabeledIntSquarer, labeledIntList, settings);
+            SQUARER_BATCHING_DESC_V2, callLabeledIntSquarer, labeledIntList, settings, EXECUTOR);
     Future<Integer> result = underTest.add(2);
     assertThat(result.isDone()).isTrue();
     assertThat(result.get()).isEqualTo(4);
   }
 
+  @Test
+  public void testWhenDelayThresholdExceeds() throws Exception {
+    BatchingSettings settings =
+        batchingSettings.toBuilder().setDelayThreshold(Duration.ofMillis(200)).build();
+    underTest =
+        new BatcherImpl<>(
+            SQUARER_BATCHING_DESC_V2, callLabeledIntSquarer, labeledIntList, settings, EXECUTOR);
+    Future<Integer> result = underTest.add(6);
+    assertThat(result.isDone()).isFalse();
+    // Give time for the delay to trigger and push the batch.
+    Thread.sleep(300);
+    assertThat(result.isDone()).isTrue();
+    assertThat(result.get()).isEqualTo(36);
+  }
+
+  @Test
+  public void testBlockingFlush() throws Exception {
+    final long sleepTimeMs = 200L;
+    BatchingDescriptor<Integer, Integer, LabeledIntList, List<Integer>> descriptor =
+        new SquarerBatchingDescriptorV2() {
+          @Override
+          public void splitResponse(
+              List<Integer> batchResponse, List<SettableApiFuture<Integer>> batch) {
+            try {
+              Thread.sleep(sleepTimeMs);
+            } catch (InterruptedException e) {
+              e.printStackTrace();
+            }
+            super.splitResponse(batchResponse, batch);
+          }
+        };
+    BatchingSettings settings =
+        batchingSettings.toBuilder().setDelayThreshold(Duration.ofMillis(50)).build();
+    underTest =
+        new BatcherImpl<>(descriptor, callLabeledIntSquarer, labeledIntList, settings, EXECUTOR);
+    long startTime = System.currentTimeMillis();
+    Future<Integer> result = underTest.add(2);
+    // Give time for the delay to trigger the batch through executor.
+    Thread.sleep(50);
+    underTest.flush();
+    long totalTime = System.currentTimeMillis() - startTime;
+    assertThat(totalTime).isAtLeast(sleepTimeMs + 50);
+    assertThat(result.isDone()).isTrue();
+  }
+
+  /** Validates that the elements are not leaking to multiple batches */
+  @Test(timeout = 800)
+  public void testElementsNotLeaking() throws Exception {
+    ListeningExecutorService multiExecutor =
+        MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(20));
+
+    final BatchTestCallable callable = new BatchTestCallable();
+    BatchingSettings settings =
+        batchingSettings.toBuilder().setDelayThreshold(Duration.ofMillis(50)).build();
+
+    try (BatcherImpl<Integer, Integer, LabeledIntList, List<Integer>> batcherTest =
+        new BatcherImpl<>(SQUARER_BATCHING_DESC_V2, callable, labeledIntList, settings, EXECUTOR)) {
+
+      final ElementAddThread single = new ElementAddThread(batcherTest);
```

Comment:
What’s single, please use more descriptive variable names

Reply:
My apologies,  I forgot to give this proper name.